subxt/src/tx/tx_client.rs

```diff
@@ -103,11 +103,14 @@ impl<T: Config, C: OfflineClientT<T>> TxClient<T, C> {
     }
 
     /// Create a partial extrinsic.
-    pub fn create_partial_signed_with_nonce<Call>(
+    ///
+    /// Note: This function will **not** refine the params to use the latest account nonce and the latest block hash/number.
+    /// Instead your call will get default values of nonce = 0 and mortality = Immortal for the signed extensions, if the params
+    /// have not been 'refined' before.
+    pub fn create_partial_signed_offline<Call>(
```

Comment:
I am not sure about the name here, maybe `create_partial_signed_unrefined`? Ideally, we would have different implementations of `create_partial_signed` for offline and online clients, but without [`specialization`](https://rust-lang.github.io/rfcs/1210-impl-specialization.html) I don't see how that would be possible.

Reply:
An alternative to specialization with stable rust is the autoref workaround. (James pointed this out a while ago and we ended up using it in substrate for metadata v15)

The `default` implementation should be implemented on the `&T`. Whenever users what to override that, they must implement it on `T`. Then the compiler will add as many references as needed to T to find the method call. Not entirely sure if this is applicable here

Reply:
Hehe it might be possible but we probably shouldn't go down that route for fear of over-complexifying :D

Reply:
I like the `_offline` suffix too FWIW, and also this suffix plays nicely with my suggested comment tweak so that it is consistent that if we call an `_offline` function we get worse defaults because it runs offline.

Reply:
Ah I remember reading dtolnays article about autoref specialization on stable rust a while ago, but it seemed like a weird workaround to me.